File: frontend/src/lib/models/campaign.model.ts
import type { CampaignStatus } from "../enums/campaign-status.enum";
import type { IStoryboard } from "./storyboard.model";

export interface ICampaign {
  id: string;
  /**
   * Associated asset ID
   */
  assetId: string;
  /**
   * Creation timestamp
   */
  createdAt: string;
  status: CampaignStatus;
  /**
   * ICampaign storyboard
   */
  storyboard?: IStoryboard;
  /**
   * Last update timestamp
   */
  updatedAt?: string;
}

File: backend/internal/database/models/campaign.model.go
package models

import "github.com/gianghp123/Vidmerce/backend/internal/core/enums"

type CampaignEntity struct {
	BaseItem
	// Associated asset ID
	AssetID string `dynamodbav:"assetId"`
	// Creation timestamp
	CreatedAt string               `dynamodbav:"createdAt"`
	Status    enums.CampaignStatus `dynamodbav:"status"`
	// CampaignEntity storyboard
	StoryboardEntity *StoryboardEntity `dynamodbav:"storyboard,omitempty"`
	// Last update timestamp
	UpdatedAt *string `dynamodbav:"updatedAt,omitempty"`
}
